backend/globaleaks/handlers/analyst/__init__.py

```diff
@@ -57,6 +66,146 @@ def get_stats(session, tid):
         "reports_tor": num_tips_tor
     }
 
+@transact
+def get_stats_fields(session, tid):
+    tid_lang = ConfigFactory(session, tid).get_val('default_language')
+    fields = session.query(models.Field).filter(
+        models.Field.statistical == True,
+        models.Field.tid == tid
+    ).all()
+    fields_dict = [{'id': x.id, 'label': x.label.get(tid_lang)} for x in fields]
+
+    aux = ['internal_tip_id', 'internal_tip_status', 'internal_tip_creation_date','internal_tip_update_date',
+           'internal_tip_expiration_date', 'internal_tip_receiver_count', 'last_access',
+           'internal_tip_creation_date_month', 'internal_tip_creation_date_years', 'internal_tip_file_count',
+           'internal_tip_comment_count'
+           ]
+    for i in aux:
+        fields_dict.append(
+            {
+                'id': i,
+                'label': i
+            }
+        )
+    return fields_dict
+
+def get_month_safe(date_value):
+    if isinstance(date_value, datetime):
+        return date_value.month
+    return 0
+
+def get_year_safe(date_value):
+    if isinstance(date_value, datetime):
+        return date_value.year
+    return 0
+
+def get_base_stats(session, internal_tip_id):
+    internal_tip = session.query(models.InternalTip).filter(
+        models.InternalTip.id == internal_tip_id
+    ).one_or_none()
+    count_comment = session.query(
+        func.count(distinct(models.Comment.id))).filter(
+        models.Comment.internaltip_id == internal_tip_id,
+        models.Comment.visibility == models.EnumVisibility.public.value
+    ).scalar()
+    count_files = session.query(func.count(distinct(models.InternalFile.id))).filter(
+        models.InternalFile.internaltip_id == internal_tip_id
+    ).scalar()
+    count_receivers = session.query(func.count(distinct(models.ReceiverTip.id))).filter(
+        models.ReceiverTip.internaltip_id == internal_tip_id
+    ).scalar()
+    return {
+        'internal_tip_id': internal_tip.id,
+        'internal_tip_status': internal_tip.status,
+        'internal_tip_creation_date': internal_tip.creation_date,
+        'internal_tip_creation_date_month': get_month_safe(internal_tip.creation_date),
+        'internal_tip_creation_date_years': get_year_safe(internal_tip.creation_date),
+        'internal_tip_update_date': internal_tip.update_date,
+        'internal_tip_expiration_date': internal_tip.expiration_date,
+        'last_access': internal_tip.last_access,
+        'internal_tip_file_count': count_files,
+        'internal_tip_comment_count': count_comment,
+        'internal_tip_receiver_count': count_receivers,
+    }
```

Comment:
This functions seems to require testing on systems with 10.000 reports (like the system for which it is designed. Making 4 queries for each report when generating does not scale. 
We expect it will be needed to rewirte this function to perform just 4 queries in total.